pageserver/src/page_cache.rs

```diff
@@ -4,22 +4,30 @@
 //! a hash map, keyed by the tenant ID.
 
 use crate::PageServerConf;
-//use crate::repository::Repository;
+use crate::repository::Repository;
+use crate::walredo::PostgresRedoManager;
 use crate::repository::rocksdb::RocksRepository;
 use lazy_static::lazy_static;
+use std::path::Path;
 use std::sync::{Arc, Mutex};
 
 lazy_static! {
-    pub static ref REPOSITORY: Mutex<Option<Arc<RocksRepository>>> = Mutex::new(None);
+    pub static ref REPOSITORY: Mutex<Option<Arc<dyn Repository + Send + Sync>>> = Mutex::new(None);
 }
 
 pub fn init(conf: &PageServerConf) {
     let mut m = REPOSITORY.lock().unwrap();
 
-    *m = Some(Arc::new(RocksRepository::new(conf)));
+    // Set up a WAL redo manager, for applying WAL records.
+    let walredo_mgr = PostgresRedoManager::new(conf);
+
+    // we have already changed current dir to the repository.
+    let repo = RocksRepository::new(conf, Path::new("."), Arc::new(walredo_mgr));
+
+    *m = Some(Arc::new(repo));
 }
 
-pub fn get_repository() -> Arc<RocksRepository> {
+pub fn get_repository() -> Arc<dyn Repository + Send + Sync> {
```

Comment:
@ericseppanen, @patins and other Rust gurus: would it make sense to include the Send and Sync in the Repository trait itself, so that we wouldn't need to spell them out like this? And how do you do that? :-)

Reply:
`trait Repository: Send + Sync { ... }`

Reply:
Thanks, changed it to that.